android/src/main/java/com/swmansion/reanimated/layoutReanimation/SharedTransitionManager.java

```diff
@@ -53,7 +60,34 @@ protected View getTransitioningView(int tag) {
     return mCurrentSharedTransitionViews.get(tag);
   }
 
-  protected void screenDidLayout() {
+  protected void screenDidLayout(int screenViewTag) {
+    ReactContext context = mAnimationsManager.getContext();
+    EventDispatcher eventDispatcher =
+        UIManagerHelper.getEventDispatcherForReactTag(context, screenViewTag);
+    if (eventDispatcher != null && !eventDispatchersWithListener.contains(eventDispatcher)) {
+      eventDispatchersWithListener.add(eventDispatcher);
+      eventDispatcher.addListener(
```

Comment:
I wonder if we could approach it differently utilizing the existing mechanism for rerounting events that we use for other type of events (gestures / scroll / etc). If I remember correctly we don't use reflection there to access individual fields. Also, the progress event need to be forwarded to the JS runtime ultimately which is something we already do for that other type of events. If I remember correctly, we use view tag and event name to register for such events, and it appears that we have all that information here.

Reply:
I tried to use the existing event mechanism, but I think that the current approach is better.

To register an event, we need the event name and the view tag of the component that emits it. So we need the screen `reactTag`. However, we can only obtain the screen tag on the UI Thread and it require a new native function. The `ReaniamtedNativeModule.registerEventHandler` method needs to be called from JavaScript Thread because  the `ReaniamtedNativeModule` isn't available on UI Thread.

There are two places where we can register for an event. The first is the mounting phase, but it's too early to obtain the view screen, so we can't use it. The other option is on transition start, but that's too late since the component is reparented to the transition container.

Reply:
From what I understand it is our code from reanimated that has these requirements. In the current approach we monitor screen layout events for all screen types. I think that we can modify the registration logic such that it allows us to monitor all type of events regardless of the tag, and only make the decision whether we want to start shared transition based on the subviews from a given screen.

From my perspective it'd feel cleaner to hardcode the name of the event rather than hardcoding private field names as we do it now